lightning/src/ln/channelmanager.rs

```diff
@@ -3950,6 +3973,33 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> MessageSend
 				result = NotifyOption::DoPersist;
 			}
 
+			if self.startup_complete.load(Ordering::Relaxed) != 0 {
+				let new_feerate = self.fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Normal);
+
+				let mut handle_errors = Vec::new();
+				{
+					let mut channel_state_lock = self.channel_state.lock().unwrap();
+					let channel_state = &mut *channel_state_lock;
+					let pending_msg_events = &mut channel_state.pending_msg_events;
+					let short_to_id = &mut channel_state.short_to_id;
+					channel_state.by_id.retain(|chan_id, chan| {
+						let counterparty_node_id = chan.get_counterparty_node_id();
+						let (retain_channel, chan_needs_persist, err) = self.update_channel_fee(short_to_id, pending_msg_events, chan_id, chan, new_feerate);
+						if chan_needs_persist == NotifyOption::DoPersist { result = NotifyOption::DoPersist; }
+						if err.is_err() {
+							handle_errors.push((err, counterparty_node_id));
+						}
+						retain_channel
+					});
+				}
+
+				for (err, counterparty_node_id) in handle_errors.drain(..) {
+					let _ = handle_error!(self, err, counterparty_node_id);
+				}
+
+				self.startup_complete.store(1, Ordering::Release);
```

Comment:
Is this line reachable? It seems like we won't hit this unless `self.startup_complete.load(Ordering::Relaxed) != 0` already